src/language.c

```diff
@@ -238,20 +239,28 @@ static void read_lang(char *langfile)
         lskip = 1;
       }
       if (lskip) {
-        while (!strchr(lbuf, '\n') && fgets(lbuf, 511, FLANG) != NULL) {
+        while (!strchr(lbuf, '\n') && fgets(lbuf, sizeof lbuf, FLANG) != NULL) {
           lline++;
         }
         /* fgets == NULL means error or empty file, so check for error */
         if (ferror(FLANG)) {
           putlog(LOG_MISC, "*", "LANG: Error reading lang file.");
         }
-        lline++;
         lnew = 1;
         continue;
       }
-      strcpy(ltext, strchr(lbuf, ',') + 1);
-    } else
-      strcpy(strchr(ltext, 0), lbuf);
+      if ((ctmp = strchr(lbuf, ',')))
+        strcpy(ltext, ctmp + 1);
+      else 
+        putlog(LOG_MISC, "*", "LANG: Malformed text line (missing ,) in %s at %d.",
+               langfile, lline);
+    } else {
+      if ((strlen(ltext) + strlen(lbuf) + 1) > ltextsize) {
+        ltextsize += sizeof lbuf;
+        ltext = nrealloc(ltext, ltextsize);
+      }
+      strcat(ltext, lbuf);
```

Comment:
strcat, really? :)

Reply:
better than what came before : strcpy(strchr(ltext, 0), lbuf);
haha.
but you are right, why not go then extra mile and do it really proper.

Reply:
Fixed.